crd-generator/api/src/main/java/io/fabric8/crd/generator/AbstractJsonSchema.java

```diff
@@ -132,85 +120,208 @@ protected T internalFrom(TypeDef definition, String... ignore) {
-      final String[] description = new String[1];
-
+      String name = property.getName();
       if (property.isStatic() || ignores.contains(name)) {
         LOGGER.debug("Ignoring property {}", name);
         continue;
       }
-      
-      property.getAnnotations().forEach(a -> {
-        switch(a.getClassRef().getFullyQualifiedName()) {
-          case "javax.validation.constraints.NotNull":
-            required.add(name);
-            break;
-          case ANNOTATION_JSON_PROPERTY:
-            updatePropertyFromAnnotationIfNeeded(property, name, updated, a);
-            break;
-          case ANNOTATION_JSON_PROPERTY_DESCRIPTION:
-            updateDescriptionFromAnnotation(name, description, a);
-            break;
-        }
-      });
 
-      final String accessorMatcher = "(is|get|set)" + property.getNameCapitalized();
-      methods.stream()
-        .filter(m -> m.getName().matches(accessorMatcher))
-        .forEach(m -> m.getAnnotations().forEach(a -> {
-          switch(a.getClassRef().getFullyQualifiedName()) {
-            case ANNOTATION_JSON_PROPERTY:
-              updatePropertyFromAnnotationIfNeeded(property, name, updated, a);
-              break;
-            case ANNOTATION_JSON_PROPERTY_DESCRIPTION:
-              updateDescriptionFromAnnotation(name, description, a);
-              break;
-          }
-        }));
+      final PropertyFacade facade = new PropertyFacade(property, accessors);
+      final Property possiblyRenamedProperty = facade.process();
+      name = possiblyRenamedProperty.getName();
 
-      final Property possiblyRenamedProperty = updated[0] != null ? updated[0] : property;
-      final T schema = internalFrom(possiblyRenamedProperty.getName(), possiblyRenamedProperty.getTypeRef());
+      if (facade.required) {
+        required.add(name);
+      }
+      final T schema = internalFrom(name, possiblyRenamedProperty.getTypeRef());
       // if we got a description from the field or an accessor, use it
+      final String description = facade.description;
       final T possiblyUpdatedSchema;
-      if (description[0] == null) {
+      if (description == null) {
         possiblyUpdatedSchema = schema;
       } else {
-        possiblyUpdatedSchema = addDescription(schema, description[0]);
+        possiblyUpdatedSchema = addDescription(schema, description);
       }
       addProperty(possiblyRenamedProperty, builder, possiblyUpdatedSchema);
     }
     return build(builder, required);
   }
 
-  private void updatePropertyFromAnnotationIfNeeded(Property property, String name, Property[] updated,
-    AnnotationRef a) {
-    if (updated[0] != null) {
-      LOGGER.debug("Property {} has already been renamed to {}", name, updated[0].getName());
-      return;
+  private Map<String, Method> indexPotentialAccessors(TypeDef definition) {
+    final List<Method> methods = definition.getMethods();
+    final Map<String, Method> accessors = new HashMap<>(methods.size());
+    methods.stream()
```

Comment:
what about inherited properties and accessors?

Reply:
Aren't the TypeDef that are passed there "complete"? Plus, this is not different from what was done before i.e. the previous implementation was only checking the TypeDef's methods for accessors. This bit only makes it slightly more efficient by avoiding to iterate over all methods for each property.

Reply:
@iocanel if you think this is an issue, I think we should open a different issue.

Reply:
`TypeDef` instances do not include properties and method available in super classes. This is why I recently added `RichTypeDef` that allows accessing to superclass properties and methods.

As long as we are aware of this, we can proceed and cover this in  a future pr.